File: app/src/main/java/com/example/order_with/menuItem/Menu.java
package com.example.order_with.menuItem;

import android.os.Parcel;
import android.os.Parcelable;

public class Menu implements Parcelable {
    private String title;
    private String price;

    public Menu(String title, String price) {
        this.title = title;
        this.price = price;
    }

    protected Menu(Parcel in) {
        title = in.readString();
        price = in.readString();
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(title);
        dest.writeString(price);
    }

    @Override
    public int describeContents() {
        return 0;
    }

    public static final Creator<Menu> CREATOR = new Creator<Menu>() {
        @Override
        public Menu createFromParcel(Parcel in) {
            return new Menu(in);
        }

        @Override
        public Menu[] newArray(int size) {
            return new Menu[size];
        }
    };

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getPrice() {
        return price;
    }

    public void setPrice(String price) {
        this.price = price;
    }

    @Override
    public String toString() {
        return "Menu{" +
                "title='" + title + '\'' +
                ", price='" + price + '\'' +
                '}';
    }
}

File: PROJECT/app/src/main/java/com/example/order_with/Data/Menu.kt
package com.example.order_with.Data

import android.os.Parcel
import android.os.Parcelable


data class Menu(
    val name: String,
    val num: Int,
    val price: String
) : Parcelable {
    constructor(source: Parcel) : this(
        source.readString().toString(),
        source.readInt(),
        source.readString().toString()
    )

    override fun describeContents() = 0

    override fun writeToParcel(dest: Parcel, flags: Int) = with(dest) {
        writeString(name)
        writeInt(num)
        writeString(price)
    }

    companion object {
        @JvmField
        val CREATOR: Parcelable.Creator<Menu> = object : Parcelable.Creator<Menu> {
            override fun createFromParcel(source: Parcel): Menu = Menu(source)
            override fun newArray(size: Int): Array<Menu?> = arrayOfNulls(size)
        }
    }
}
